diff --git a/lib/system.c b/lib/system.c
index b851110..34e519b 100644
--- a/lib/system.c
+++ b/lib/system.c
@@ -387,7 +387,7 @@ add_system_trust(gnutls_x509_trust_list_t list,
 #ifdef DEFAULT_TRUST_STORE_FILE
 	ret =
 	    gnutls_x509_trust_list_add_trust_file(list,
-						  DEFAULT_TRUST_STORE_FILE,
+						  getenv("SSL_CERT_FILE") || DEFAULT_TRUST_STORE_FILE,
 						  crl_file,
 						  GNUTLS_X509_FMT_PEM,
 						  tl_flags, tl_vflags);
